pytorch_lightning/overrides/base.py

```diff
@@ -46,6 +46,13 @@ def forward(self, *inputs, **kwargs):
 
         if running_stage == RunningStage.TRAINING:
             output = self.module.training_step(*inputs, **kwargs)
+
+            # In manual_optimization, we need to prevent DDP reducer as
+            # it is done manually in ``LightningModule.manual_backward``
+            # `require_backward_grad_sync` will be reset in the
+            # ddp_plugin ``post_training_step`` hook
+            if not self.module.automatic_optimization:
+                self.module.trainer.model.require_backward_grad_sync = False
```

Comment:
self.module.trainer.model is quite a few hops :/ not sure how we can make it better though.

Reply:
We thought about that as well. Unfortunately we need to get on between the lightning module and the ddp wrapper here. 
The only other option we found would be to patch the usercode. So we think that this option is the better one.